pages/api/account/settings/[id].ts

```diff
@@ -0,0 +1,50 @@
+import type { NextApiResponse, NextApiRequest } from "next"
+import sierraClient from "../../../../src/server/sierraClient"
+import initializePatronTokenAuth from "../../../../src/server/auth"
+
+/**
+ * API route handler for /api/account/settings/{patronId}
+ */
+export default async function handler(
+  req: NextApiRequest,
+  res: NextApiResponse
+) {
+  let responseMessage
+  let responseStatus
+  const patronTokenResponse = await initializePatronTokenAuth(req.cookies)
+  const cookiePatronId = patronTokenResponse.decodedPatron?.sub
+  if (!cookiePatronId) {
+    responseStatus = 403
+    responseMessage = "No authenticated patron"
+    return res.status(responseStatus).json(responseMessage)
+  }
+  if (req.method == "PUT") {
+    /**  We get the patron id from the request: */
+    const patronId = req.query.id as string
+    const patronData = req.body
+    /**  We check that the patron cookie matches the patron id in the request,
+     * i.e.,the logged in user is updating their own settings. */
+    if (patronId == cookiePatronId) {
+      const response = await settingsUpdate(patronId, patronData)
+      responseStatus = response.status
+      responseMessage = response.message
+    } else {
+      responseStatus = 403
+      responseMessage = "Authenticated patron does not match request"
+    }
+  }
+  res.status(responseStatus).json(responseMessage)
```

Comment:
`responseMessage` and `responseStatus` should have default values. Try this:
- log in so your `cookiePatronId` is valid
- hit the API endpoint in your browser
- this gets returned:
![Screen Shot 2024-02-23 at 3 20 25 PM](https://github.com/NYPL/research-catalog/assets/1280564/00d84cdd-0a56-4b22-aa8f-cafa6d818737)

We _don't_ expect this endpoint to be hit through a GET request but it would be nice, to have a complete API, to return a better error message. This can be either a default status/message or you can check if the method is a GET request and return a "please make a post request....." message.